### PR TITLE
main/dhcp: improve

### PR DESCRIPTION
**Pull request by tcely**
Fixes https://bugs.alpinelinux.org/issues/10161
Fixes https://bugs.alpinelinux.org/issues/10024

Resolves alpinelinux/aports#6265
Resolves alpinelinux/aports#6435

### CODE REVIEW

**Comment by tcely** [comment]
Incorporated changes from https://github.com/alpinelinux/aports/pull/6435

File: main/dhcp/APKBUILD [hunk 1]
```diff
@@ -28,49 +34,138 @@ source="http://ftp.isc.org/isc/$pkgname/${_realver}/$pkgname-$_realver.tar.gz
...
+
+	make DESTDIR="$pkgdir" install
+
 	#install -m751 -D "$builddir"/client/scripts/linux "$pkgdir"/sbin/dhclient-script
 	install -m751 -D "$srcdir"/dhclient-script-alpine "$pkgdir"/sbin/dhclient-script
 	install -d "$pkgdir"/var/lib/dhcp
 	install -d "$pkgdir"/var/run/dhcp
 }
 
+dbg() {
```
Comment (Ikke) [inline main/dhcp/APKBUILD]: Why are you overriding this? `depends_dbg` is undefined. Is the reason to reset the dependencies?

Comment (tcely) [reply main/dhcp/APKBUILD]: Yes, so that the `depends` from `dhcp` do not apply to the `dhcp-dbg` apk too.

Comment (Ikke) [reply main/dhcp/APKBUILD]: Would it perhaps be better to either explicitly make it empty, or define `depends_dbg` (and other similar variables) as empty?

Comment (tcely) [reply main/dhcp/APKBUILD]: I'm happy to include empty variables at the top for any that are currently unset.

**Comment by tcely** [comment]
> I have no right to push to main, but I can review it.

It'd be nice to be able to squash these pull requests to a single branch on GitHub, then let the people who have permissions to push to main deal with the commits from that branch.

**Comment by tcely** [comment]
@nmeum can we get these fixes merged?

**Comment by ncopa** [comment]
It looks like `dhcrelay-openrc` gets installed with either dhcp server package but not with `dhcrelay`.

````
ncopa-desktop:~/aports/main/dhcp$ sudo apk add dhcp-server-ldap
(1/4) Installing dhcp (4.4.1-r3)
Executing dhcp-4.4.1-r3.pre-install
(2/4) Installing dhcp-openrc (4.4.1-r3)
(3/4) Installing dhcrelay-openrc (4.4.1-r3)
(4/4) Installing dhcp-server-ldap (4.4.1-r3)
Executing busybox-1.30.1-r1.trigger
OK: 5683 MiB in 1527 packages
ncopa-desktop:~/aports/main/dhcp$ sudo apk del dhcp-server-ldap
(1/4) Purging dhcp-openrc (4.4.1-r3)
(2/4) Purging dhcrelay-openrc (4.4.1-r3)
(3/4) Purging dhcp (4.4.1-r3)
(4/4) Purging dhcp-server-ldap (4.4.1-r3)
Executing busybox-1.30.1-r1.trigger
OK: 5667 MiB in 1523 packages
ncopa-desktop:~/aports/main/dhcp$ sudo apk add dhcrelay
(1/1) Installing dhcrelay (4.4.1-r3)
Executing busybox-1.30.1-r1.trigger
OK: 5669 MiB in 1524 packages
````

**Pull request closed without merging**